Add Instance Plans as aliases

edward-murrell opened the pull request:
This adds the ability select plans by a shortname, rather than a integer ID.

## Description
The VC2 plan is identified by VCPU, RAM, and Storage.

For example:
```hcl
resource "vultr_server" "server0" {
  plan_id = module.vultr_lib.vc1_m1_s1x25
}
```

### Checklist:

* :heavy_check_mark: Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* :heavy_check_mark: Have you linted your code locally prior to submission?
* :heavy_check_mark: Have you successfully ran tests with your changes locally?

ddymko commented:
@edward-murrell 

Thanks for taking the time to do this! 

We do have one request could we change the naming scheme? Right now it looks like we are blending multiple instance types VC2, VHF (High Frequency), and VDC (Dedicated)

I am thinking it might be better to break them out into their own sets


https://api.vultr.com/v1/plans/list?type=vc2
```
Vultr Cloud Compute (VC2)

- vc2-512mb-1vcpu-ipv6
- vc2-512mb-1vcpu

- vc2-1vcpu-1gb
- vc2-1vcpu-2gb
- vc2-2vcpu-4gb
- vc2-4vcpu-8gb
- vc2-6vcpu-16gb
- vc2-8vcpu-64gb
- vc2-16vcpu-64gb
- vc2-24vcpu-97gb
```

https://api.vultr.com/v1/plans/list?type=vhf
```
Vultr High Frequency (VHF)

- vhf-1vcpu-1gb
- vhf-1vcpu-2gb
- vhf-2vcpu-4gb
- vhf-3vcpu-8gb
- vhf-4vcpu-16gb
- vhf-8vcpu-32gb
- vhf-12vcpu-48gb
```

https://api.vultr.com/v1/plans/list?type=vdc2
```
Vultr Dedicated Cloud (VDC)

- vdc-4vcpu-8gb
- vdc-4vcpu-16gb
- vdc-6vcpu-24gb
- vdc-8vcpu-32gb
```

There is also baremetal which has a different API call
https://api.vultr.com/v1/plans/list_baremetal
```
Vultr Bare Metal (VBM)

- vbm-4cpu-32gb
```

edward-murrell commented:
@ddymko Yep, that's not a big deal. I'll probably write a python script to build this automatically, and that could become part of a CI process to build these files automatically in the future.

It looks like the 512mb / ipv6 instances are only available via the UI though? They don't show up in the list requested from the API.

edward-murrell commented:
@ddymko I've added those VM plans, including the baremetal plan.

The plan aliases are now generated from the attached python script, and the `plan_id` is referenced directly. A similar system could be used to generate various other aliases in this module.

Since the http lookup at the top of the file doesn't do anything, so I've removed that from `plans.tf`.

ddymko commented:
@Oogy mind taking a look?

ddymko commented:
@edward-murrell`512mb / ipv6 instances` Yup these are only available via the UI.

This looks good!

Also, fwiw we are working on a V2 of the API that's where the naming convention for plans came from (regions also have a renaming).

Oogy commented:
@edward-murrell 
Could we have the output names changed to use underscores instead of hyphens to keep with Terraform best practices? Other than that looks great and thank you for the script.

edward-murrell commented:
@ddymko That'd be excellent, I look forward to using v2 then.

@Oogy You're welcome.

No problem at all. Done. I've changed the hyphens to underscores in the script and the plans files.
